blueocean-dashboard/src/main/js/components/InputParameters.jsx

```diff
@@ -0,0 +1,214 @@
+        originalParameters[index].defaultParameterValue.value = event;
+        this.setState({ parameters: originalParameters });
+    }
+
+    /**
+     * Creates an array from the parameter object which is in the current state
+     * @returns array - of values
+     */
+    stateParametersToArray() {
+        const { parameters } = this.state;
+        return Object.keys(parameters).map(key => {
+            const item = parameters[key];
+            return { name: item.name, value: item.defaultParameterValue.value };
+        });
+    }
+
+    /**
+     * Generic submit function. The calculations for the url has been done in
+     * @see createFormState Here we simply POST the data to the server.
+     * @param body - could be ok or cancel body
+     */
+    submitForm(body) {
+        const { href } = this.state;
+        const fetchOptions = {
+            credentials: 'include',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(body),
+        };
+        return isoFetch(href, fetchOptions)
+            .then(
+                response => {
+                    if (response.status >= 300 || response.status < 200) {
+                        const error = new Error(response.statusText);
+                        error.response = response;
+                        throw error;
+                    }
+                    return response.json();
+                }
+            );
+    }
+
+    /**
+     * Submit the form as "cancel"
+     */
+    cancelForm() {
+        this.refs.startWrapper.hide();
+    }
+
+    /**
+     * Submit the form as "ok" out of the state data parameters.
+     */
+    okForm() {
+        const body = { parameters: this.stateParametersToArray() };
+        this.submitForm(body)
+            .then((runInfo) => ToastUtils
+              .createRunStartedToast(this.props.runnable, runInfo, this.props.onNavigation));
+    }
+
+    show() {
+        this.refs.startWrapper.show();
+    }
+
+    render() {
+        console.log(this.props);
+        const { parameters } = this.state;
+        // Early out
+        if (!parameters) {
+            return null;
+        }
+        const message = t('parameterized.pipeline.header', { defaultValue: 'Pipeline parameter' });
+        const ok = t('parameterized.pipeline.submit', { defaultValue: 'Build' });
+        const parametersArray = Object.keys(parameters).map(key => parameters[key]);
+
+        // console.log('state', this.state);
+        // console.log('stateToFormSubmit', this.stateParametersToArray());
+        const cancelCaption = t('rundetail.input.cancel');
+        const cancelButton = (<a title={cancelCaption} onClick={() => this.cancelForm()} className="btn inputStepCancel run-button btn-secondary" >
+            <span className="button-label">{cancelCaption}</span>
+        </a>);
+        return (<ModalView
```

Comment:
@scherler is this the dialog component that @sophistifunk built or a new impl of the Modal? I believe we should be using the dialog not a new component (cc @brody)